test/quantization/quantize_/workflows/int8/test_int8_tensor.py

```diff
@@ -0,0 +1,218 @@
+        self.assertEqual(result_q8.shape, reference.shape)
+        self.assertTrue(compute_error(result_fp, reference) > 10)
+        self.assertTrue(compute_error(result_q8, reference) > 10)
+
+    def test_dynamic_quantization(self):
+        """Test dynamic activation quantization"""
+        weight_q8_dynamic = Int8Tensor.from_hp(
+            self.weight_fp,
+            self.block_size,
+            act_quant_kwargs=QuantizeTensorToInt8Kwargs(),
+        )
+
+        reference = torch.nn.functional.linear(self.input_fp, self.weight_fp, self.bias)
+        result_dynamic = torch.nn.functional.linear(
+            self.input_fp, weight_q8_dynamic, self.bias
+        )
+
+        self.assertEqual(result_dynamic.shape, reference.shape)
+
+    @unittest.skip("granularity parameter not supported in current API")
+    @common_utils.parametrize("granularity", [PerTensor(), PerRow()])
+    def test_slice_preserves_aliasing(self, granularity):
+        config = Int8DynamicActivationInt8WeightConfig(
+            granularity=granularity, version=2
+        )
+        l = torch.nn.Linear(1024, 1024).to("cuda").to(torch.bfloat16)
+        l.weight = torch.nn.Parameter(
+            torch.zeros(1024, 1024, dtype=torch.bfloat16, device="cuda")
+        )
+        quantize_(l, config)
+        param = l.weight
+        param_data = param.data
+        param_data = param_data.narrow(0, 0, 512)
+        # Making sure the aliasing is preserved in sliced quantized Tensor
+        assert param.data.qdata.data_ptr() == param_data.qdata.data_ptr()
+        assert param.data.scale.data_ptr() == param_data.scale.data_ptr()
+
+    @common_utils.parametrize(
+        "config",
+        [
+            Int8DynamicActivationInt8WeightConfig(version=2),
+            Int8WeightOnlyConfig(version=2),
+        ],
+    )
+    @common_utils.parametrize("device", ["cpu", "cuda"])
+    @common_utils.parametrize("dtype", [torch.bfloat16, torch.float16])
+    def test_slice(self, config, device, dtype):
+        """Test tensor slicing"""
+        dummy = torch.nn.Linear(256, 256, bias=False, dtype=dtype, device=device)
+        quantize_(dummy, config)
+
+        weight1 = dummy.weight.clone().narrow(0, 0, 64)
+        weight2 = dummy.weight.clone().narrow(1, 0, 128)
+
+        self.assertEqual(weight1.qdata, dummy.weight.qdata.narrow(0, 0, 64))
+        self.assertEqual(weight2.qdata, dummy.weight.qdata.narrow(1, 0, 128))
+
+    def test_transpose(self):
+        """Test transpose operation"""
+        weight_q8 = Int8Tensor.from_hp(self.weight_fp, self.block_size)
+        transposed = weight_q8.transpose(0, 1)
+
+        self.assertEqual(transposed.shape, (3, 4))
+        self.assertEqual(transposed.block_size, [3, 4])
+
+    def test_select(self):
+        """Test select operation"""
+        weight_q8 = Int8Tensor.from_hp(self.weight_fp, self.block_size)
+        selected = weight_q8.select(0, 0)
+
+        self.assertEqual(selected.shape, (3,))
```

Comment:
test the data as well?

Reply:
you can follow this: https://github.com/pytorch/ao/blob/5346f0e169c58aeba04076ace2f99e6b4f158102/test/quantization/quantize_/workflows/float8/test_float8_tensor.py#L449